tdrs-backend/tdpservice/users/serializers.py

```diff
@@ -203,6 +205,316 @@ def update(self, instance, validated_data):
+    has_pending_regions_change = serializers.SerializerMethodField()
+    has_pending_feature_flags_change = serializers.SerializerMethodField()
+
+    class Meta(UserProfileSerializer.Meta):
+        """Metadata."""
+
+        # Add the new fields to the existing fields list
+        fields = UserProfileSerializer.Meta.fields + [
+            'create_change_requests',
+            'has_pending_first_name_change',
+            'has_pending_last_name_change',
+            'has_pending_regions_change',
+            'has_pending_feature_flags_change',
+            'has_fra_access',
+        ]
+
+    def validate_regions(self, value):
+        """Validate regions field."""
+        """Ensure that the regions are valid."""
+        if not value:
+            raise serializers.ValidationError(_('Regions cannot be empty.'))
+        if not isinstance(value, list):
+            raise serializers.ValidationError(_('Regions must be a list of IDs.'))
+        return value
+
+    def get_has_pending_first_name_change(self, obj):
+        """Check if there's a pending change request for first_name."""
+        return obj.has_pending_change_for_field('first_name')
+
+    def get_has_pending_last_name_change(self, obj):
+        """Check if there's a pending change request for last_name."""
+        return obj.has_pending_change_for_field('last_name')
+
+    def get_has_pending_regions_change(self, obj):
+        """Check if there's a pending change request for regions."""
+        return obj.has_pending_change_for_field('regions')
+
+    def get_has_pending_feature_flags_change(self, obj):
+        """Check if there's a pending change request for feature flags."""
+        return obj.has_pending_change_for_field('feature_flags')
+
+    def _handle_regions(self, validated_data, instance, pending_request=None):
+        """Handle the regions field specifically for change requests."""
+        change_requests = []
+        new_regions = validated_data['regions']
+        current_regions = set(instance.regions.all().values_list('id', flat=True))
+        new_region_ids = set(region.id for region in new_regions)
+
+        if pending_request:
+            if current_regions != new_region_ids:
+                # Update the existing pending request
+                pending_request.requested_value = list(new_region_ids)
+                pending_request.save()
+            change_requests.append(pending_request)
+
+        # New request
+        elif current_regions != new_region_ids:
+            # Store as a list of IDs
+            change_request = instance.request_change(
+                field_name='regions',
+                requested_value=list(new_region_ids),
+                current_value=list(current_regions),
+                requested_by=self.context['request'].user
+            )
+            change_requests.append(change_request)
+
+        return change_requests
+
+    def _handle_has_fra_access(self, validated_data, instance, pending_request=None):
```

Comment:
is there a way to make this more generic? we'd need a function for every permission with this implementation